vidyut-prakriya/tests/integration/kashika_8_4.rs

```diff
@@ -265,6 +294,164 @@ fn sutra_8_4_23() {
+    t.assert_has_jhi(&["antar"], &han, Lat, &["antarGnanti"]);
+    t.assert_has_krdanta(&["antar"], &han, Krt::lyuw, &["antarhanana"]);
+}
+
+#[test]
+fn sutra_8_4_25() {
+    let mut t = Tester::with_rule_choices(vec![RuleChoice {
+        rule: Rule::Ashtadhyayi("8.4.25"),
+        decision: Decision::Accept,
+    }]);
+    // antaH (antarasmin?) ayanam
+    t.assert_has_saptami_tatpurusha("antar", "ayana", &["antarayaRa"]);
+
+    // 8.4.25: Decline ("Desh" case)
+    t = Tester::with_rule_choices(vec![RuleChoice {
+        rule: Rule::Ashtadhyayi("8.4.25"),
+        decision: Decision::Decline,
+    }]);
+    // antaH Iyate asmin deSe
+    t.assert_has_bahuvrihi("antar", "ayana", &["antarayana"]);
+}
+
+// 8.4.[26,27] ... No tests
+
+#[test]
+fn sutra_8_4_28() {
+    assert_has_bahuvrihi("pra", "nas", &["praRas"]);
+    assert_has_bahuvrihi("ut", "nas", &["unnas"]);
+}
+
+#[test]
+fn sutra_8_4_29() {
+    let ya = d("yA", Adadi);
+    let ma = d("mA\\N", Juhotyadi);
+    assert_has_krdanta(&["pra"], &ya, Krt::lyuw, &["prayARa"]);
+    assert_has_krdanta(&["pari"], &ya, Krt::lyuw, &["pariyARa"]);
+    assert_has_krdanta(&["pra"], &ma, Krt::lyuw, &["pramARa"]);
+    assert_has_krdanta(&["pari"], &ma, Krt::lyuw, &["parimARa"]);
+}
+
+#[test]
+fn varttika_8_4_29_1() {
+    // 8.4.29.1  nirviRRasyopasaNKyAnam
+    assert_has_krdanta(&["nir"], &d("vi\\da~\\", Divadi), Krt::kta, &["nirviRRa"]);
+}
+
+#[test]
+fn sutra_8_4_30() {
+    let ya = nic(&d("yA", Adadi));
+    assert_has_krdanta(&["pra"], &ya, Krt::lyuw, &["prayApaRa", "prayApana"]);
+    assert_has_krdanta(&["pari"], &ya, Krt::lyuw, &["pariyApaRa", "pariyApana"]);
+    // Todo(tbdasap) Add the rest of the kashika examples
+}
+
+#[test]
+fn sutra_8_4_31() {
+    let kup = &d("kupa~", Divadi);
+    assert_has_krdanta(&["pra"], &kup, Krt::lyuw, &["prakopaRa", "prakopana"]);
+    assert_has_krdanta(&["pari"], &kup, Krt::lyuw, &["parikopaRa", "parikopana"]);
+    assert_has_krdanta(&["pra"], &d("Iha~", Bhvadi), Krt::lyuw, &["prehaRa"]);
+    assert_has_krdanta(&["pra"], &d("Uha~", Bhvadi), Krt::lyuw, &["prohaRa"]);
+    assert_has_krdanta(&["pra"], &d("wuvapa~", Bhvadi), Krt::lyuw, &["pravapaRa"]);
+}
+
+#[test]
+fn sutra_8_4_32() {
+    let iki = &d("iKi~", Bhvadi);
+    assert_has_krdanta(&["pra"], &iki, Krt::lyuw, &["preNKaRa"]);
+    assert_has_krdanta(&["pra"], &d("magi~", Bhvadi), Krt::lyuw, &["pramaNgana"]);
+}
+
+#[test]
+fn sutra_8_4_33() {
+    assert_has_krdanta(
+        &["pra"],
+        &d("Risi~\\", Adadi),
+        Krt::lyuw,
+        &["praRiMsana", "praniMsana"],
+    );
+    assert_has_krdanta(
+        &["pra"],
+        &d("Rikza~", Bhvadi),
+        Krt::lyuw,
+        &["praRikzaRa", "pranikzaRa"],
+    );
+    assert_has_krdanta(
+        &["pra"],
+        &d("Ridi~", Adadi),
+        Krt::lyuw,
+        &["praRindana", "pranindana"],
+    );
+}
+
+#[test]
+fn sutra_8_4_34() {
+    assert_has_krdanta(&["pra"], &d("BA\\", Adadi), Krt::lyuw, &["praBAna"]);
```

Comment:
I find it more readable to move reused dhatus into a var so I can see at a glance that tests are all using the same dhatu, e.g.:

```
let bhaa = d("BA\\", Adadi);
// then, use bhaa in the tests
```